main/src/ca/uwaterloo/flix/runtime/interpreter/Channel.java

```diff
@@ -55,6 +55,11 @@
    */
   private final boolean unbuffered;
 
+  /**
+   * a random number generator used to pick one of the ready channels of a 'select' statement.
+   */
+  private static final Random RANDOM = new Random(); // TODO: Q: Should we supply a seed?
```

Comment:
I think we should use the following:

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html

(See the description on that page and on the doc for Random).

Reply:
I.e. remove the field and use the sniplet I added below :)